akka-actor/src/main/scala/akka/pattern/RetrySettings.scala

```diff
@@ -0,0 +1,149 @@
+import scala.jdk.DurationConverters.JavaDurationOps
+
+final case class RetrySettings(
+    maxRetries: Int,
+    delayFunction: Int => Option[FiniteDuration],
+    shouldRetry: Throwable => Boolean = _ => true) {
+
+  override def toString: String = {
+    s"RetrySettings(maxRetries=$maxRetries, delayFunction=$delayFunction)"
+  }
+
+  def withExponentialBackoff(
+      minBackoff: FiniteDuration,
+      maxBackoff: FiniteDuration,
+      randomFactor: Double): RetrySettings =
+    RetrySettings(maxRetries, minBackoff, maxBackoff, randomFactor)
+
+  def withExponentialBackoff(minBackoff: Duration, maxBackoff: Duration, randomFactor: Double): RetrySettings =
+    RetrySettings.create(maxRetries, minBackoff, maxBackoff, randomFactor)
+
+  def withDelay(fixedDelay: FiniteDuration): RetrySettings =
+    RetrySettings(maxRetries, fixedDelay)
+
+  def withDelay(fixedDelay: Duration): RetrySettings =
+    RetrySettings.create(maxRetries, fixedDelay)
+
+  def withDelay(delayFunction: Int => Option[FiniteDuration]): RetrySettings =
+    copy(delayFunction = delayFunction)
+
+  def withDelayFunction(delayFunction: java.util.function.IntFunction[Optional[java.time.Duration]]): RetrySettings = {
+    val scalaFunc: Int => Option[FiniteDuration] = attempt => {
+      val javaDuration = delayFunction.apply(attempt)
+      if (javaDuration.isPresent)
+        Some(javaDuration.get.toScala)
+      else
+        None
+    }
+    withDelay(scalaFunc)
+  }
+
+  def withDecider(shouldRetry: Throwable => Boolean): RetrySettings =
+    copy(shouldRetry = shouldRetry)
+
+  def withDeciderFunction(shouldRetry: java.util.function.Function[Throwable, Boolean]): RetrySettings =
+    copy(shouldRetry = shouldRetry.apply)
+}
+
+object RetrySettings {
+
+  def apply(maxRetries: Int): RetrySettings = {
+    // Start with a reasonable minimum backoff (e.g., 100 milliseconds)
+    val minBackoff: FiniteDuration = 100.millis
+
+    // Max backoff increases with maxRetries, capped to a sensible upper limit (e.g., 1 minute)
+    val maxBackoff: FiniteDuration = if (maxRetries > 10) { //to avoid multiplication overflow
+      1.minute
+    } else {
+      val base = minBackoff * math.pow(2, maxRetries).toLong
+      base.min(1.minute)
+    }
+
+    // Random factor can scale slightly with maxRetries to add more jitter
+    val randomFactor: Double = (0.1 + (maxRetries * 0.05)).min(1.0) // cap at 1.0
+
+    apply(maxRetries, minBackoff, maxBackoff, randomFactor)
+  }
+
+  def create(maxRetries: Int): RetrySettings = {
+    apply(maxRetries)
+  }
+
+  def apply(maxRetries: Int, fixedDelay: FiniteDuration): RetrySettings = {
+    RetrySettings(maxRetries, FixedDelayFunction(fixedDelay))
+  }
+
+  def create(maxRetries: Int, fixedDelay: Duration): RetrySettings = {
+    apply(maxRetries, fixedDelay.toScala)
+  }
+
+  def apply(
+      maxRetries: Int,
+      minBackoff: FiniteDuration,
+      maxBackoff: FiniteDuration,
+      randomFactor: Double): RetrySettings =
```

Comment:
> keep in mind that we have a sort of creation dualism here

yeah, seems unnecessary to have both factory methods and builder methods. Then we are back to zillions of overloaded methods again. I'd remove the factory methods (apply/create) and only leave the one with single parameter `maxRetries: Int`

Then used as:
```
RetrySettings(5).with...
```

Reply:
the one with `config` can also be kept

Reply:
done